BaseClasses.py

```diff
@@ -113,6 +113,11 @@ def extend(self, regions: Iterable[Region]):
             for region in regions:
                 self.region_cache[region.player][region.name] = region
 
+        def add_group(self, new_id: int):
+            self.region_cache.update({new_id: {}})
+            self.entrance_cache.update({new_id: {}})
+            self.location_cache.update({new_id: {}})
```

Comment:
you should really just create the key explicitly. While this isn't a performance critical codepath admittedly, this is still creating and then unwrapping another dict for no reason.